### PR TITLE
openmp-17: new version stream for openmp

### PR DESCRIPTION
**Pull request by ajayk**

### CODE REVIEW

File: openmp-17.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      - cmake
+      - samurai
+      - binutils-dev
+      - libffi-dev
+      - zlib-dev
+      - libxml2-dev
+      - pkgconf
+      - python3
+      - help2man
+      - llvm17
```
Comment (hectorj2f) [inline openmp-17.yaml:25]: Do we need both llvm17 and llvm17-dev?

File: openmp-17.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  version: 17.0.5
+  epoch: 0
+  description: "LLVM OpenMP library"
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - wolfi-base
```
Comment (hectorj2f) [inline openmp-17.yaml:12]: nit: Wolfi-base already includes the `ca-certificates-bundle` in case you want to simplify it.

File: openmp-17.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      - pkgconf
+      - python3
+      - help2man
+      - llvm17
+      - llvm-cmake-17
+      - llvm17-dev
+      - libLLVM-17
+
+pipeline:
+  - uses: fetch
```
Comment (hectorj2f) [inline openmp-17.yaml:31]: suggestion: we could use git-checkout here, which sounds like the ideal choice when possible.

File: openmp-17.yaml (hunk 1 above)
Comment (ajayk) [reply openmp-17.yaml:25]: yes we need both 

llvm17 provides `usr/lib/llvm17/bin/llvm-tblgen` which is needed here

File: openmp-17.yaml (hunk 2 above)
Comment (ajayk) [reply openmp-17.yaml:12]: wolf-base doesn't include  `ca-certificates-bundle` , its build-base which provides  `ca-certificates-bundle` 
 build-base -> wolfi-baselyout -> ca-certificates-bundle

Comment (hectorj2f) [reply openmp-17.yaml:12]: question: do we care about adding a package if another package already includes it ?

**Review by hectorj2f — APPROVED** [review]

File: openmp-17.yaml (hunk 2 above)
Comment (ajayk) [reply openmp-17.yaml:12]: I think we should remove them , its spread across various packages , there might be some reasoning behind it , i am not completely sure

File: openmp-17.yaml (hunk 3 above)
Comment (ajayk) [reply openmp-17.yaml:31]: will address this in the upcoming pr

**Pull request merged**